### PR TITLE
Increasing the scope of the TSC and Incubator.

### PR DESCRIPTION
**Pull request by mikeal**
After today's TSC meeting I sat down and attempted to write something that might define the scope of what we would or would not admit into the incubator. The more I tried to clarify that language the more I stop believing it could actually be done.

Here's the problem: If we are going to increase the scope of the TSC to supporting the ecosystem, which is to say we are opening it up to helping the broader community, then the scope and priorities of what to admit need to be dictated by that community and not whoever happens to be in the TSC at any given point in time. Whatever we could write in an attempt to define a scope will be incorrect because it will be driven by our own views of what the ecosystem is or needs rather than on what that community is actually asking for and saying that it needs.

All I believe we can do is outline what we can't do and outline how we will protect the foundation and the incubator from abuse or how we might protect any resources we have that are finite.

I imagine this will kick off a whole new discussion now about what the scope of the TSC and Incubator should be, so here's my personal thoughts:
- Proliferating open governance strengthens the resiliency of our community and, ultimately, of Node.js as a technology and a solution everywhere it is adopted. We have collected a surprising amount of experience in how to do this and know that there is no "one size fits all" approach, it's about values and mentorship.
- For a variety of reasons projects need a "home." We have very little control over when they decide they need that home but we do know, from some of our own experiences on the matter, that when the contributors/users decide they need a neutral home it's hard if not impossible to make progress without one. We could be that home and, provided we can do so at little to no cost to our existing liabilities, we probably should be.
- For better or worse we're now who people are looking to in the Node.js community for guidance. To me that means we have a responsibility to respond and offer our support because this is our community after all.

I'm not saying we should "grow just to grow." I'm saying that we should grow to be as big as our community is asking us to be. We have a fair amount of infrastructure that we've built and I'm having a very hard time figuring out exactly what we lose by using it to support _any_ part of our community that asks for it.

### CODE REVIEW

**Comment by mikeal** [comment]
Following up a little more.

The TSC will never be more than a relatively small representation of the Node.js community and ecosystem. The community and ecosystem are huge and constantly growing in scope. The TSC is never going to be the experts on what the ecosystem needs or how to prioritize which projects deserve support and which don't and so I don't see much of a way to prioritize other than to say "if they are applying then they obviously think they need our support."

However, I **do** think that the TSC will be the experts on what the current constraints and limitations are on what the foundation can provide to projects and any resources we have that are limited or under too much stress.

With that in mind all I think we can do is iterate on better ways to quantify any limits in resources and to more accurately capture during the application process the needs of each project asking to be admitted.

File: Project-Lifecycle.md [hunk 1]
```diff
@@ -98,6 +98,8 @@ vote to abstain from representation on the TSC.
 
 ## Applying to join
 
+The Node.js Foundation only admits projects written in Node.js.
```
Comment (dshaw) [inline Project-Lifecycle.md]: Node.js, libuv, http-parser are all not "written in Node.js" (whatever that means).

Comment (TheAlphaNerd) [inline Project-Lifecycle.md]: perhaps "written in Node.js or supporting the Node.js ecosystem" is better?

Comment (mikeal) [inline Project-Lifecycle.md]: Good point.

**Comment by jasnell** [comment]
Good to see this come in so quickly... :-) .. unfortunately won't have time to review until tomorrow.

File: README.md [hunk 2]
```diff
@@ -2,10 +2,43 @@
 
 The Node.js Foundation Technical Steering Committee is the technical governing body of the Node.js Foundation. It admits and oversees all top-level Projects in the Node.js Foundation. It also elects a representative to the Node.js Foundation Board of Directors.
 
+The TSC is community and contributor driven and is a home for community managed 
+projects and assets. The mission of the Node.js Foundation is to enable and encourage
```
Comment (orangemocha) [inline README.md]: Referring to the TSC as a "home" for "projects and assets" sounds slightly confusing to me. Did you mean the Foundation rather than the TSC? Since most of the language is repeated later in this paragraph, I would skip this sentence here and begin the paragraph with "The mission of the Node.js Foundation..."

File: README.md [hunk 3]
```diff
@@ -2,10 +2,43 @@
 
 The Node.js Foundation Technical Steering Committee is the technical governing body of the Node.js Foundation. It admits and oversees all top-level Projects in the Node.js Foundation. It also elects a representative to the Node.js Foundation Board of Directors.
 
+The TSC is community and contributor driven and is a home for community managed 
+projects and assets. The mission of the Node.js Foundation is to enable and encourage 
+the adoption of Node.js and the TSC is the home of the community driven portion
+of executing that mission.
+
```
Comment (orangemocha) [inline README.md]: +100 for "The mission of the Node.js Foundation is to enable and encourage the adoption of Node.js"

Instead of:
> the TSC is the home of the community driven portion of executing that mission

how about:
> the TSC is the body responsible for executing that mission on behalf of the community

File: README.md [hunk 4]
```diff
@@ -2,10 +2,43 @@
...
+Since the Foundation and the TSC are intended to support the community the size and scope
+of the work and projects in it will grow to accomodate the needs of the community.
+
```
Comment (orangemocha) [inline README.md]: This statement assumes that growing the size of the size and scope of the work and projects is the only / best way to support the community. It also implies unconditional growth of the scope as a mission of the TSC. 

As we discussed previously, diluting the limited resources is one of the potential drawbacks of such unconditional growth. 

Not objecting to growth in principle but I think that the passage from the *what* (the mission of the TSC, fostering the Node.js ecosystem) to the *how* needs to be further elaborated and this statement may fail to capture the complexity of the criteria involved.

I would either leave it out, or soften it a bit, for example to something like  "it *may* grow to accommodate the needs of the community, within the constraints of available resources and in accordance to the priorities determined by the TSC"

Comment (orangemocha) [inline README.md]: `accomodate` => `accommodate`

File: README.md [hunk 5]
```diff
@@ -2,10 +2,43 @@
...
+* Provide a neutral "home" for projects so that they are not owned by a single individual or company.
+* Mentor projects in the Node.js community on open governance and liberal contribution policies.
+* Ensure that the Node.js Foundation does not charter projects that do not adhere to the values 
+of the Node.js Foundation. Projects must be: Transparent, Participatory and Effective.
+* The TSC does not "pick winners" and may admit multiple projects that serve similar use cases. 
+Admitting a project for incubation or even chartering it is not an endorsement by the Node.js 
+Foundation or TSC on the technical merits of a project, it is only an endorsement that the project 
+adheres to the TSC's values.
+
+Other than being written in Node.js there are no technical requirements on projects to be admitted
```
Comment (orangemocha) [inline README.md]: As already suggested, being written in Node.js should not be a requirement, so "Other than being written in Node.js" should be removed.

File: README.md (hunk 2 above)
Comment (mikeal) [inline README.md]: This is a little complicated in terms of phrasing.

The technical community driven part of the foundation is managed by the TSC. To say something is "in the Node.js Foundation" when it is a technical project managed by its contributors and community that means it is under the TSC and managed on the the technical side of the foundation.

The "incubator" is how we admit projects, but to say "the incubator is home to projects" is only part of the story because eventually projects graduate from the incubator but are still under the TSC.

But yes, this is hard to articulate well and I'm open to suggestions for better wording.

File: README.md (hunk 4 above)
Comment (mikeal) [inline README.md]: > This statement assumes that growing the size of the size and scope of the work and projects is the only / best way to support the community. It also implies unconditional growth of the scope as a mission of the TSC. 

That's not an assumption I'm trying to make. There's a difference here to the TSC being *reactive* and growing to meet demands of projects in the ecosystem and the TSC being *proactive* and growing "just to grow."

There's plenty the foundation is doing to support the community completely outside of the TSC. I think there's a false impression here that the TSC needs to remove resources from Core or proactively create new resources in order to be a home to projects that are asking for support. I don't believe that is the case, I think that we can admit projects at little to no cost to the current work we're doing. But the pertinent question here how we respond to projects *asking* us for this support. Do we tell them that we have some abstract criteria based on our own personal priorities that limits what kinds of projects we'll allow in or do we simply try to respond to what the community is asking for.

> As we discussed previously, diluting the limited resources is one of the potential drawbacks of such unconditional growth. 

What exactly are these resources? Other than mentorship, which is primarily on my shoulders early on, what exactly are we diluting? What resources are we allocating to these projects that we're in danger of running out of?

> Not objecting to growth in principle but I think that the passage from the what (the mission of the TSC, fostering the Node.js ecosystem) to the how needs to be further elaborated and this statement may fail to capture the complexity of the criteria involved.

Here's the dilemma, how do we figure out what the Node.js ecosystem needs other than responding to what they are asking for? How do we establish the *how* of "how to support the ecosystem" other than to admit those that apply? We are not representative enough of the ecosystem to build objective criteria to what they need and the ecosystems is constantly changing and its needs may change as well.

File: README.md (hunk 5 above)
Comment (mikeal) [inline README.md]: Yup, working on that.

File: README.md (hunk 4 above)
Comment (orangemocha) [inline README.md]: Ok, I see that you didn't intend to make the assumption though I interpreted it that way. I would suggest rewording "will grow to accommodate the needs of the community" to "may grow as needed to meet the needs of the community".

>What exactly are these resources? Other than mentorship, which is primarily on my shoulders early on, what exactly are we diluting? What resources are we allocating to these projects that we're in danger of running out of?

Even without us having collected a full list, I would assume that resources are not infinite :) I could guess: CI resources, marketing dollars, TSC staff time. I think we will learn this more clearly as we try to grow the scope.

> How do we establish the how of "how to support the ecosystem" other than to admit those that apply? 

How about inviting representative of potential member projects to the TSC meetings for some time, so we have a chance to learn from the perspectives of the community?

Comment (mikeal) [inline README.md]: > Even without us having collected a full list, I would assume that resources are not infinite :) I could guess: CI resources, marketing dollars, TSC staff time. I think we will learn this more clearly as we try to grow the scope.

We'll certainly continue to identify more areas where we are limited and can taper accepting applications in accordance with those resources. However, all the things you listed aren't even entitlements projects receive :)

**CI resources:** This is under the CTC and core. The current Build WG is *for* core and those resource aren't allocated to incoming projects. Additional projects will have their own resources and, if they need more, will build up their own competency similar to the CTC Build WG.

**Markting dollars:** The foundation doesn't market *core* they market Node.js and that already includes relevant projects written in Node.js inside or outside of the foundation whenever relevant. In fact, having more projects under the same roof only helps them bring in the right people to help with that messaging when necessary.

**TSC staff time:** The TSC is not allocated to incoming projects, incoming projects are given a *mentor(s)* which isn't necessarily a TSC member, the list of mentors is on the README and is currently only a single person (me). This will grow over time but the reason it is separate from the TSC is so that we can grow the list of members wider than the TSC membership. Other than that the only thing the TSC will be spending time on is evaluating applications and eventually charters once projects are ready to graduate the incubator.

Every project is different, it's possible that some projects will have needs we haven't even considered or identified yet. The only productive step I can see us taking is adding an "Infrastructure" section to applications and working closely with applicants to flush it out and identify all the necessary resource allocations, if any, but we shouldn't be surprised if there is little if nothing they need other than a proper home.

 > How about inviting representative of potential member projects to the TSC meetings for some time, so we have a chance to learn from the perspectives of the community?

There's two problems with this. The first is that the TSC membership has a narrow view of the community as a whole. None of us really understand how wide the community is and could invite the right people. The second problem is that we are unlikely to find the areas of the community *in need* of our support this way. Some projects already appear to need some support from the foundation while others seem perfectly fine without us, but that is *our* interpreation of what is going on and the reality may be very different. The only way we can respond to what the community actually needs is to let them tell us what they need by applying.

**Comment by jasnell** [comment]
This LGTM with the change already in works for the "written in Node.js" bit :-)

File: README.md [hunk 6]
```diff
@@ -2,10 +2,43 @@
 
 The Node.js Foundation Technical Steering Committee is the technical governing body of the Node.js Foundation. It admits and oversees all top-level Projects in the Node.js Foundation. It also elects a representative to the Node.js Foundation Board of Directors.
 
+The TSC is community and contributor driven and is a home for community managed 
+projects and assets. The mission of the Node.js Foundation is to enable and encourage 
+the adoption of Node.js and the TSC is the home of the community driven portion
```
Comment (rvagg) [inline README.md]: I don't think we need the second half of this sentence, it's telling the TSC to stay out of anything not "community driven", which is vague but also not in the spirit of how the Foundation is set up. Although it may be in the spirit of how the Foundation currently operates, with two strictly divided parts, I don't believe this is optimal or in Node's best interests.

> The mission of the Node.js Foundation is to enable and encourage the adoption of Node.js.

Ought to be enough to guide the TSC.

File: README.md [hunk 7]
```diff
@@ -2,10 +2,43 @@
...
+of executing that mission.
+
+Since the Foundation and the TSC are intended to support the community the size and scope
```
Comment (rvagg) [inline README.md]: Well... no, you said above what the mission is, "support the community" is not in that mission. The use of "community" throughout this document may be unhelpful given how difficult it is to define the term, it's nice to use when you handwave but this ought not be a handwavy document, we should attempt to find more concrete terms if we are going this direction.

"The size and scope of the work governed by the TSC will grow as is required in order to responsibly meet the goals of the Node.js Foundation."

File: README.md [hunk 8]
```diff
@@ -2,10 +2,43 @@
...
+## Projects, Working Groups and the Incubator
+
+The Node.js Foundation is setup to be the home of multiple autonomous projects. Projects are 
+admitted into the incubator and are mentored until their governance and contributor 
+base reach a state of maturity. They are then given a proper charter and a seat on the TSC.
+
+The TSC also forms Working Groups to tackle tasks that don't scope to a single project. These 
+Working Groups can be given a seat on the TSC at the TSC's discretion.
+
+The goal of the Incubator is to:
```
Comment (rvagg) [inline README.md]: I propose we restructure this section, "Node.js Ecosystem" is another vague term that's not helpful in this context. Suggestion:

--------------------------------

The goal of the Incubator is to assist the TSC as it seeks to enable and encourage Node.js
adoption by providing a clear on-boarding path for the addition of new top-level projects.

The Incubator process includes:

* Mentoring of projects on open governance and liberal contribution policies as used in other
  Node.js Foundation projects.
* Ensuring that the Node.js Foundation does not charter projects that do not support the mission
  of enabling and encourage Node.js adoption.
* Ensuring that the Node.js Foundation does not charter projects that do not adhere to the values
  of the Node.js Foundation as represented by other Node.js Foundation projects. This is
  currently understood to mean that proects ought to be: Transparent, Participatory and Effective.

The TSC does not "pick winners" and may admit multiple projects that serve similar use cases. 
Admitting a project for incubation or even chartering it as a top-level project is not an endorsement
by the Node.js Foundation or TSC on the technical merits of a project, it is only an endorsement
that the project adheres to Node.js Foundation governance models, values and supports the
mission of the Node.js Foundation in some way.

--------------------------------

Further, I like this but I don't think it belongs in "Incubator" because that's not supposed to be a "home", it belongs in top-level projects somewhere:

> Provide a neutral "home" for projects so that they are not owned by a single individual or company.

**Comment by rvagg** [comment]
The text seems to be a little at odds with some of your statements @mikeal, I wonder if your comments here are reflecting a change in opinion about how this should be approached? Aside from my comments inline, I'm leaning +1 on the changes.

What I'm hearing from you though is that you don't think the TSC is in a very good position to "support the ecosystem" because it's shaped by "whoever happens to be in the TSC at any given time". Perhaps specifically that right now it's core-focused which is why we're being a little resistant to expansion and embracing the projects lining up. I think you're suggesting that instead to be a mechanism whereby expansion would be "dictated by that community", by being open-by-default to new projects.

This seems pretty arbitrary to me, basically saying that expansion should be directed by whoever shows up rather than trying to form an objective process as a group. Projects are most likely to show up because someone encourages them to show up, that's how these things tend to work (and are currently working in our case). I'd also call for patience so that we're all moving forward with enough confidence that we can be all working on the same goals.

While it's true that the TSC is currently very core-focused and this may be frustrating because we seem to be having difficulty thinking beyond core, please remember that the Foundation was set up around core and it's the logical center of technical activity in pursuing the mission of _enabling and encouraging adoption of Node.js_. Also when you look at the CTC, the Collaborator group and everyone contributing, notice that the overwhelming majority of them either were focused somewhere outside of core (in the "community" / "ecosystem") or still are. In fact, many of us as individuals and the companies behind us, flocked to core precisely to support the adoption of Node.js out of concern over the risk that the state of core posed for that adoption!

So in my opinion, even though it may be frustrating to work through a group process to make this happen, the TSC, as a collective, is best placed to make a judgement on what projects should receive help in order to pursue the mission of the Foundation. This should not be the role of the Board and absolutely not the role of any lone individual (Foundation staff or otherwise), nor can it be left to just accepting _whoever shows up at the door_ unless we come up with a really good reason why not. It's only through a diverse group process that we will discover the best path to achieving that mission and set up a sustainable and objective process that is (relatively) free from abuse.

**Comment by mikeal** [comment]
@rvagg sure, right now the TSC is very core focused. But even if we change that in the future and it's representative of many ecosystem projects, it's still not going to be "representative of the entire community." I'm thinking back over the last 5 years and remembering how many things grew without most of us noticing. Huge areas of growth that we didn't even notice until they were already bigger than areas we were focused on.

When I think about answering the question "What can the TSC/Foundation do for the community?" I don't have an answer I'm comfortable executing on other than "What they tell us they need."

This is not to say that I don't want to have a process, as you suggested earlier, I just think that this process should be focused entirely on managing any finite resources we have and not trying to form opinions and strategy on what *types* of projects we're taking. Who are we to say some particular class of projects is more important or more in need of our support than another? We are historically bad predictors of this ;)

I think that, as a mission statement, we should support the community to the best of our ability. In practice and process we should strive to do the following:

* Identify what resources we offer to projects and what the availability, or lackthereof, is of those resources.
* Refine the application process to better identify the needs of projects that are applying.
* Clearly message the benefits of joining the incubator as well as what we **do not** offer and are the responsibility of the project to build/provide.

Like all of our processes this is something we'll have to iterate on and get better at over time, but I think the goals of the admission process should primarily be to identify our own limited resources and manage them effectively.

There's another part of this that I think is worrying to the TSC and comes through in these comments: what is protecting the TSC/Foundation from be *used* by bad actors? To some extent a lot of our concerns about "picking a winner" are related to this as well.

While we have a good process in place for ensuring a project can't leave the incubator as a project supported by a single company there is a possibility that just entering the incubator could serve some competitive advantage to projects. In this I have very few good ideas on how to manage or protect us. The best idea I have is to try and build up more Top Level Working Groups that represent a sub-community (databases, web frameworks, IoT) and once they are established we could ask for their help in vetting incoming projects.

File: README.md (hunk 8 above)
Comment (mikeal) [inline README.md]: I like this text better as well, I'll incorporate it in my next push.

File: README.md (hunk 7 above)
Comment (mikeal) [inline README.md]: hrm... I don't think we can avoid using the word community entirely. while "contributor driven" describes part of the story we also are driven but the users of the work we do and make great effort to incorporate the needs of those users above the whims of contributors.

File: README.md (hunk 6 above)
Comment (mikeal) [inline README.md]: I agree that this is poorly worded but I do think we need something that defines the separation.

Sure, the TSC may from time to time get involved in the Marketing Committee or the Board but it is doing so with the mandate that it represents the contributors and the community.

File: README.md [hunk 9]
```diff
@@ -2,10 +2,43 @@
...
+* Mentor projects in the Node.js community on open governance and liberal contribution policies.
+* Ensure that the Node.js Foundation does not charter projects that do not adhere to the values 
+of the Node.js Foundation. Projects must be: Transparent, Participatory and Effective.
+* The TSC does not "pick winners" and may admit multiple projects that serve similar use cases. 
+Admitting a project for incubation or even chartering it is not an endorsement by the Node.js 
+Foundation or TSC on the technical merits of a project, it is only an endorsement that the project 
+adheres to the TSC's values.
+
+Other than being written in Node.js there are no technical requirements on projects to be admitted
+into the Node.js Foundation. However, because resources may be limited from time to the TSC may
```
Comment (tjwebb) [inline README.md]: > because resources may be limited from time to the TSC

[grammar] "from time to time"

**Comment by tjwebb** [comment]
I really like the spirit of this discussion and the proposed amendments. (thanks @mikeal for the link; it's nice to see some familiar faces as well. What's up @TheAlphaNerd? :smile:)

Some thoughts I'd like to offer on the current discussion:

> there is a possibility that just entering the incubator could serve some competitive advantage to projects

I'm considering submitting an application to the foundation to consider accepting one of my projects. To the degree that "adoption" by the Foundation imbues the project a level of integrity in its governance that it did not enter with, the goals of the Foundation and the project are not in conflict. For those of us who see open and consistent project governance as a major asset to the project and its users, I think we'd see this as a major benefit of applying to the Foundation.

> There's another part of this that I think is worrying to the TSC and comes through in these comments: what is protecting the TSC/Foundation from be used by bad actors? To some extent a lot of our concerns about "picking a winner" are related to this as well.

One thing that occurs to me is that using the active rhetorical voice at all implies a level of commitment/endorsement in an individual project that the Foundation maybe doesn't intend; by this, I mean that instead of saying that the Foundation "admits" certain projects into it, it might be worth exploring wording that instead illustrates the opposite flow: that perhaps the Foundation "joins" or" collaborates with" the project as an advisor or mentor. I can see an argument that the effect of the current of wording may end up having a stronger effect than anticipated; that is, it may have the effect of signaling a level of intentional selection that confers a sort of specialness to the project that the Foundation doesn't actuallu confer. I'm not sure if this would help clarify or not, but I see it as a possible way to distance the Foundation from any perceived endorsement of the project's technical or populist bonafides.

File: README.md [hunk 10]
```diff
@@ -2,10 +2,43 @@
 
 The Node.js Foundation Technical Steering Committee is the technical governing body of the Node.js Foundation. It admits and oversees all top-level Projects in the Node.js Foundation. It also elects a representative to the Node.js Foundation Board of Directors.
 
+The TSC is community and contributor driven and is a home for community managed 
+projects and assets. The mission of the Node.js Foundation is to enable and encourage 
+the adoption of Node.js and the TSC is the home of the community driven portion
+of executing that mission.
```
Comment (jasnell) [inline README.md]: Perhaps something like:

```
It is the mission of the Node.js Foundation and the TSC to provide a home for 
openly-governed and community-managed open source projects that are focused on 
supporting and providing benefit to the entire Node.js ecosystem.
```

File: README.md [hunk 11]
```diff
@@ -2,10 +2,43 @@
...
+
 For more details read the [TSC Charter](https://github.com/nodejs/TSC/blob/master/TSC-Charter.md) adopted by the Node.js Foundation Board of Directors on June 17th 2015.
 
 If your project is interested in joining the Node.js Foundation please read the [Project Lifecycle.md](./Project-Lifecycle.md) documentation.
 
+## Projects, Working Groups and the Incubator
+
+The Node.js Foundation is setup to be the home of multiple autonomous projects. Projects are 
+admitted into the incubator and are mentored until their governance and contributor 
+base reach a state of maturity. They are then given a proper charter and a seat on the TSC.
```
Comment (jasnell) [inline README.md]: Perhaps: 

```
The Node.js Foundation is structured to serve as a home for multiple autonomous projects. Projects are first admitted into an incubation phase where they are mentored until the project governance and contributor base reaches a state of maturity. The TSC then works with the project to establish an official charter and a seat on the TSC.
```

File: README.md (hunk 8 above)
Comment (jasnell) [inline README.md]: Perhaps: 
```
The goal of the Incubator is to:

* Support the ecosystem of Node.js users and contributors by providing a 
  home for openly governed and community managed projects built on, or 
  built to support Node.js.
* Mentor projects within the Node.js community on open governance and liberal 
  contribution policies.
* Ensure that the Node.js Foundation does not charter projects that do not 
  adhere to the values upon which the Node.js Foundation was established. 
  Projects must be: Transparent, Participatory and Effective.
* Ensure that the Node.js Foundation and the TSC does not "pick a winner" in 
  the ecosystem by equally and fairly mentoring multiple projects that serve 
  similar use cases. Admitting project's for incubation, or chartering mature 
  projects, is not an endorsement by the Node.js Foundation or TSC on the 
  technical merits of a project; rather, it is an endorsement that the project 
  adheres to the Node.js Foundation and TSC's open governance values.

There are no technical requirements on projects to be admitted into the Node.js 
Foundation incubator. The projects will, however, be evaluated on various 
criteria including alignment with Node.js Foundation goals, support of and for 
the Node.js ecosystem, and relevance of scope to Node.js development. 

Because resources may be limited from time to the TSC may elect to delay 
admission until resources become available.
```

**Comment by rvagg** [comment]
@dougwilson part of the problem is that this is all fairly new so we don't have clear definitions of these things, effectively they are describing the "spirit" of how we work but as yet I doubt we have anything close enough to being a solid codified way of defining these. If the entire CTC and the Collaborators of Node.js were to disappear tomorrow and a new group be transplanted in, I doubt that they would be able to use our documentation to replicate how we work.

So part of the TSC's role over time will be document these things. Mentorship is a mechanism for spreading the values and mechanisms that we currently find useful but we also ought not limit ourselves to these. It's likely that new projects coming in would clarify things about the model and help us define the mechanics but it's also likely that the new projects would execute on these values in different ways and explore new mechanics for achieving similar goals.

So right now, it's unfortunate, but it's basically using those words in the current, but broad, first-world, English-speaking, western sense and saying that these are the values that we want to ensure that all projects share and that there is no room for projects that want to operate in ways that may conflict with these.

I know that's not really an answer, but that's the way things are now and if we were to attempt to document clear procedures and policies that we expect to be developed then we'd probably not complete the exercise for all the bikeshedding involved. And in some ways, it's a bit hypocritical for us to force heavy top-down policies when what we are really trying to achieve is bottom-up, facilitated from the top. We just don't quite yet know what the boundaries are that we should be imposing from the top to make this thing sustainable beyond the individuals from whom these values come.

**Comment by jasnell** [comment]
@rvagg +1 ... to add to this: if you look at the actual process for landing commits and making changes in core, it's actually far more organic and bottoms up in reality. The key goal on the governance side is to iteratively capture the spirit of what we're working to achieve as opposed to *imposing* process constraints. Nothing is really set in stone, everything is open to iteration, and much of process is being figured out as we go as we discover or evolve requirements through experience. 

As far as accepting new projects is concerned, in my view it comes down to three things: (1) we don't want the foundation to pick a winner, (2) we want to foster projects that follow the spirit of the bottoms-up, open collaboration, lazy-consensus model that Node.js core uses, and (3) we want to support/sponsor projects that directly support, and have the support of, the community as a whole. Niche projects or projects that are only peripherally or marginally related to node, or that are not truly community supported, or are too tightly controlled by any one party are likely not going to be a good fit.

**Comment by mikeal** [comment]
@dougwilson so, as far as the "Transparent, Participator and Effective" definition, these are not absolutes.

What we want is for projects to seek the right balance of these values, not to consider each one an absolute. And we also recognize that each project is different and is going to need a different balance of these.

The best example here is the difference between how the Node.js Website Working Group operates and how Node.js Core operates. Node Core has to merge most master commits through several branches for different release lines which gets incredibly difficult without nice squashed merge commits and proper metadata. That makes it a bit harder for newcomers to contribute but it's a necessary tradeoff in order to be effective. The website doesn't have that constraint, master is just deployed, so the project is much more liberal about merges and commit messages. There are dozens of similar examples of these tradeoffs across the working groups and other projects.

What most foundations do is create a process that embodies their values and force every project to adhere to the same process. We've almost entirely avoided that and instead try to clearly layout the values and work with projects through mentorship on finding the best project specific processes that work.

**Comment by mikeal** [comment]
Ok, I *think* my last push dealt with all the specific textual issues people had but, honestly, there's been so much discussion I may have missed some :/

File: README.md [hunk 12]
```diff
@@ -2,10 +2,56 @@
 
 The Node.js Foundation Technical Steering Committee is the technical governing body of the Node.js Foundation. It admits and oversees all top-level Projects in the Node.js Foundation. It also elects a representative to the Node.js Foundation Board of Directors.
 
+It is the mission of the Node.js Foundation and the TSC to provide a home for 
+openly-governed and community-managed open source projects that are focused on 
+supporting and providing benefit to the entire Node.js ecosystem.
```
Comment (csaintamant) [inline README.md]: Is the use of the word "entire" here intentionally setting a very high bar for projects? It's doubtful there are many projects outside of Node core itself that could be interpreted as supporting the *entire* Node.js ecosystem. In fact, I'm hard pressed to come up with a single project that fits this definition.

File: README.md [hunk 13]
```diff
@@ -2,10 +2,56 @@
...
+
+* Support the ecosystem of Node.js users and contributors by providing a 
+  home for openly governed and community managed projects built on, or 
+  built to support Node.js.
+* Mentor projects within the Node.js community on open governance and liberal 
+  contribution policies.
+* Ensure that the Node.js Foundation does not charter projects that do not 
+  adhere to the values upon which the Node.js Foundation was established. 
+  Projects must be: Transparent, Participatory and Effective.
+* Ensure that the Node.js Foundation and the TSC does not "pick a winner" in
```
Comment (csaintamant) [inline README.md]: This seems like a fairly important factor, especially in terms of the perception of the first few projects admitted. A deeply-buried clause _saying_ that [admitting projects] is "not an endorsement by the Node.js Foundation or TSC" may not have much impact on people's perception of the foundation picking a winner.

File: README.md [hunk 14]
```diff
@@ -2,10 +2,56 @@
...
+  home for openly governed and community managed projects built on, or 
+  built to support Node.js.
+* Mentor projects within the Node.js community on open governance and liberal 
+  contribution policies.
+* Ensure that the Node.js Foundation does not charter projects that do not 
+  adhere to the values upon which the Node.js Foundation was established. 
+  Projects must be: Transparent, Participatory and Effective.
+* Ensure that the Node.js Foundation and the TSC does not "pick a winner" in 
+  the ecosystem by equally and fairly mentoring multiple projects that serve 
+  similar use cases. Admitting project's for incubation, or chartering mature
```
Comment (csaintamant) [inline README.md]: Typo here, "project's" should be plural, not possessive.

File: README.md (hunk 13 above)
Comment (mikeal) [inline README.md]: Agreed, but we can also message this when we announce projects entering the foundation. The important thing is that people entering the incubator and the TSC understand this commitment so that we can speak to it consistently.

File: Project-Lifecycle.md (hunk 1 above)
Comment (orangemocha) [inline Project-Lifecycle.md]: This line is still incorrect.

File: README.md [hunk 15]
```diff
@@ -2,10 +2,56 @@
 
 The Node.js Foundation Technical Steering Committee is the technical governing body of the Node.js Foundation. It admits and oversees all top-level Projects in the Node.js Foundation. It also elects a representative to the Node.js Foundation Board of Directors.
 
+It is the mission of the Node.js Foundation and the TSC to provide a home for 
+openly-governed and community-managed open source projects that are focused on 
+supporting and providing benefit to the Node.js ecosystem.
```
Comment (orangemocha) [inline README.md]: IMO this is an even stronger statement than before that "the" only mission of the TSC and the Foundation is to grow the number of hosted projects. I was more comfortable with the previous version "The mission of the Node.js Foundation is to enable and encourage the adoption of Node.js".
That hosting any project that provides benefit to the Node.js ecosystems should be the mission of the TSC is still something that needs to be proven and qualified, in my opinion.

Comment (mikeal) [inline README.md]: To "enable and encourage the adoption" isn't actionable enough. The entire foundation is doing that, the question is "how is the TSC doing that?" and specifically "what is the TSC doing outside of hosting Node.js core to do that?"

The "neutral home for owning assets" part of this is becoming important to some projects. It's something that we needed when we started the foundation, and you can see similar needs popping up in the community. I don't think that we can debate whether or not there are projects that need that, it's clear they do because they are asking for it, all we can debate is whether or not that home should be under the TSC. The purpose of this PR is to have that debate.

**Comment by rvagg** [comment]
Suggesting that we pull out core values and foundational principles and refer here to a separate document containing those. I'd like to try and work on this and will attempt to find time tomorrow to propose something that lets all of these pieces fit together nicely and tell a coherent story about what the Foundation is about for TLPs and provides guidance for the TSC when making decisions (obviously to be interpreted through the lens of the individual but that's the beauty of the group process!).

File: README.md (hunk 15 above)
Comment (orangemocha) [inline README.md]: > To "enable and encourage the adoption" isn't actionable enough.

That's because it's a mission statement, not an action plan. If we try to capture the entire plan in one sentence we are bound to grossly oversimplify it, especially at this early stage. For example the fact that a project is "focused on supporting and providing benefit to the Node.js ecosystem" doesn't mean that it is of *strategic* importance to the ecosystem, but this statement make accepting all of those projects *the only* mission of the TSC, regardless of their strategic importance. 

I am not saying that the TSC shouldn't host projects beyond core. But I think that it shouldn't jump to hosting just *any* project that needs a home either, at least without a controlled learning phase and not before having clarified the criteria. Until those criteria are defined and the TSC feels comfortable about them, I would prefer a more vague mission statement to a misleading one.

**Comment by mikeal** [comment]
> I'd like to try and work on this and will attempt to find time tomorrow to propose something that lets all of these pieces fit together nicely

Agreed that this repo is a bit of a mess. It's hard to figure out which documents are which and how they fit together.

File: README.md (hunk 15 above)
Comment (Fishrock123) [inline README.md]: I quite strongly agree with @orangemocha's sentiment here.

File: Project-Lifecycle.md (hunk 1 above)
Comment (Fishrock123) [inline Project-Lifecycle.md]: I think just "Supporting the Node.js Ecosystem" should be sufficient. Anything else is a poor use of our resources since it's not actually moving _Node_ forward.

Supporting core is also supporting the ecosystem so that doesn't need to be stated.

**Comment by jasnell** [comment]
Closing due to lack of forward movement on this. Can reopen if necessary
